nilearn/decomposition/canica.py

```diff
@@ -146,20 +147,28 @@ def fit(self, imgs, y=None, confounds=None):
             This parameter is passed to nilearn.signal.clean. Please see the
             related documentation for details
         """
+        # Fit MultiPCA without sorting
+        sorted = self.sorted
+        self.sorted = False
```

Comment:
Granted my suggestion was stupid. 

It is not very clear to me why do you need to set sorted to False, then fit MultiPCA and then set it back to whatever it was ?